docs/OpenSSLSupport.md

```diff
@@ -148,6 +148,31 @@ Note that the code snippet for the RDRAND engine opt-in is required to use not o
 but also other OpenSSL APIs that internally depend on the RAND APIs. Alternatively, developers
 can implement their own RAND method to replace the default method via `RAND_set_rand_method` API.
 
+## Security Guidance of using OpenSSL APIs
+
+Few OpenSSL APIs allow users to configure an application to load files from the host filesystem.
+Such behaviors may have security implications given that the host filesystem is untrusted under the
+threat model of OE. For these reasons, OE recommends users not to use those APIs and
+put them into `deprecations.h` as part of OpenSSL headers that are shipped along with the release package.
+OE's default compiler settings explicitly include the header such that users get warnings when an
+enclave uses any of these APIs.
```

Comment:
```suggestion
OpenSSL provides APIs that allow users to configure sensitive settings like certificate trust and cryptosuite preference from files.
Because the host file system is considered untrusted in contexts such as SGX enclaves, OE SDK marks these APIs as deprecated to discourage their use.
OE SDK does this by including a "deprecations.h" file as part of the default cmake/pkgconfig build configuration,
which will produce compiler warnings when any of these APIs are used in an enclave.
The warnings can be disabled by specifying the `OE_DISABLE_OPENSSL_DEPRECATIONS` option to the compiler.
```